### PR TITLE
OGM-1473 Make our pom files more consistent with WildFly and JBoss parent pom files

### PR DESCRIPTION
**Pull request by gsmet**
* https://hibernate.atlassian.net/browse/OGM-1473

As you can imagine, it was a lot of fun :).

It's the exact same work I did on Search so Search and OGM will be consistent once merged. I still have to do this work on HV.

Let's wait for CI to run as I didn't test all the Neo4j combinations.

### CODE REVIEW

File: documentation/manual/src/main/asciidoc/modules/infinispan.asciidoc [hunk 1]
```diff
@@ -2203,7 +2203,7 @@ in any NoSQL store supported by Infinispan:
 
 How to configure it? Here is a simple cheat sheet to get you started with this type of setup:
 
-* Add `org.infinispan:infinispan-directory-provider:{infinispanVersion}` to your dependencies
+* Add `org.infinispan:infinispan-directory-provider:{infinispan-version}` to your dependencies
```
Comment (gsmet) [inline documentation/manual/src/main/asciidoc/modules/infinispan.asciidoc]: This one was a leftover of a previous cleanup.

File: core/pom.xml [hunk 2]
```diff
@@ -140,7 +140,6 @@
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
                     <forkMode>once</forkMode>
-                    <skipExec>${skipUnitTests}</skipExec>
```
Comment (gsmet) [inline core/pom.xml]: We should use the traditional `-DskipTests` instead.

**Comment by DavideD** [comment]
Thanks @gsmet 
Great work!

It's really cool to see a cleaned-up POM

**Comment by DavideD** [comment]
Merged https://github.com/hibernate/hibernate-ogm/pull/1019

**Pull request merged**